programs/rodeos/cloner_plugin.cpp

```diff
@@ -36,9 +36,10 @@ struct cloner_config : ship_client::connection_config {
 };
 
 struct cloner_plugin_impl : std::enable_shared_from_this<cloner_plugin_impl> {
-   std::shared_ptr<cloner_config>  config = std::make_shared<cloner_config>();
-   std::shared_ptr<cloner_session> session;
-   boost::asio::deadline_timer     timer;
+   std::shared_ptr<cloner_config>                                           config = std::make_shared<cloner_config>();
+   std::shared_ptr<cloner_session>                                          session;
+   boost::asio::deadline_timer                                              timer;
+   std::optional<std::function<void(const char* data, uint64_t data_size)>> streamer = {};
```

Comment:
`std::function` is implicitly optional since it can hold an empty function which results in `std::bad_function_call`. Really as you have it now you need to check that that `std::optional` is set and that the `std::function` are both set.